crates/symbolicator/src/services/symbolication.rs

```diff
@@ -303,52 +279,23 @@ struct ModuleListBuilder {
 
 impl ModuleListBuilder {
     fn new(
-        mut cfi_caches: CfiCacheModules,
-        minidump_process_state: &ProcessState,
-        executable_type: ObjectType,
+        cfi_caches: CfiCacheModules,
+        modules: Vec<(Option<CodeModuleId>, RawObjectInfo)>,
     ) -> Self {
-        // Firstly mark modules for which we needed CFI information during unwinding.  This
-        // will insert entries into cfi_caches for modules which were missing.
-        for thread in minidump_process_state.threads() {
-            for frame_pair in thread.frames().windows(2) {
-                if let [prev_frame, next_frame] = frame_pair {
-                    if let Some(code_id) = prev_frame.module().and_then(|m| m.id()) {
-                        match next_frame.trust() {
-                            FrameTrust::Scan => {
-                                cfi_caches.mark_scanned(code_id);
-                            }
-                            FrameTrust::CFIScan => {
-                                cfi_caches.mark_scanned(code_id);
-                                cfi_caches.mark_used(code_id);
-                            }
-                            FrameTrust::CFI => {
-                                cfi_caches.mark_used(code_id);
-                            }
-                            _ => (),
-                        }
-                    }
-                }
-            }
-        }
-
         // Now build the CompletedObjectInfo for all modules
-        let cfi_caches: BTreeMap<CodeModuleId, CfiModule> = cfi_caches.finalize();
-        let mut inner: Vec<(CompleteObjectInfo, bool)> = minidump_process_state
-            .modules()
+        let cfi_caches = cfi_caches.into_inner();
+
+        let mut inner: Vec<(CompleteObjectInfo, bool)> = modules
             .into_iter()
-            .map(|code_module| {
-                let mut obj_info: CompleteObjectInfo =
-                    object_info_from_minidump_module(executable_type, code_module).into();
+            .map(|(id, raw_info)| {
+                let mut obj_info: CompleteObjectInfo = raw_info.into();
 
                 // If we loaded this module into the CFI cache, update the info object with
                 // this status.
-                if let Some(code_id) = code_module.id() {
+                if let Some(code_id) = id {
                     match cfi_caches.get(&code_id) {
                         None => {
-                            // If it was not picked up from by initial
-                            // get_referenced_modules nor marked as scanned during
-                            // stackwalking, it can only be unused.
-                            obj_info.unwind_status = Some(ObjectFileStatus::Unused);
+                            obj_info.unwind_status = None;
```

Comment:
why no longer unused?

Reply:
I kind-of piggyback on the `None` to raising errors about files being missing.